From 8159a10df959638004bae19cd108f855180d00f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 7 Mar 2013 14:30:54 +0000
Subject: [PATCH] Related: rhbz#902884 check for GetSelectedMasterPage failure
 anyway

backtrace doesn't indicate that its NULL however, but the rest of
our GetSelectedMasterPage check for NULL so fix that anyway

Change-Id: I9e0b15f4a18523ff1e787ef230de1ca227aa2980
---
 sd/source/ui/toolpanel/controls/MasterPagesSelector.cxx | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sd/source/ui/toolpanel/controls/MasterPagesSelector.cxx b/sd/source/ui/toolpanel/controls/MasterPagesSelector.cxx
index 512fd1d..58fa05f 100644
--- a/sd/source/ui/toolpanel/controls/MasterPagesSelector.cxx
+++ b/sd/source/ui/toolpanel/controls/MasterPagesSelector.cxx
@@ -553,8 +553,11 @@ void MasterPagesSelector::Execute (SfxRequest& rRequest)
         case SID_TP_EDIT_MASTER:
         {
             using namespace ::com::sun::star;
-            uno::Reference<drawing::XDrawPage> xSelectedMaster (
-                GetSelectedMasterPage()->getUnoPage(), uno::UNO_QUERY);
+            uno::Reference<drawing::XDrawPage> xSelectedMaster;
+            SdPage* pMasterPage = GetSelectedMasterPage();
+            assert(pMasterPage); //rhbz#902884
+            if (pMasterPage)
+                xSelectedMaster = uno::Reference<drawing::XDrawPage>(pMasterPage->getUnoPage(), uno::UNO_QUERY);
             SfxViewFrame* pViewFrame = mrBase.GetViewFrame();
             if (pViewFrame != NULL && xSelectedMaster.is())
             {
-- 
1.8.1.2

